src/components/common/ColonyMembers/MembersFilter/MembersFilter.tsx

```diff
@@ -46,18 +45,19 @@ const MembersFilter = ({ handleFiltersCallback, isRoot }: Props) => {
     <>
       <hr className={styles.divider} />
       <Form
-        initialValues={{
+        defaultValues={{
           memberType: MemberType.All,
           verificationType: VerificationType.All,
         }}
         onSubmit={() => {}}
-        enableReinitialize
       >
-        {({ resetForm, values }: FormikProps<FormValues>) => {
+        {({ formState: { isSubmitting, isValid }, getValues, reset }) => {
+          const values = getValues();
           const showReset =
             values.verificationType !== VerificationType.All ||
             values.memberType !== MemberType.All;
           handleFiltersCallback(values);
```

Comment:
This line is causing a state change in the parent while this component is rendering: 

![err](https://user-images.githubusercontent.com/64402732/223082620-3ebd54d4-bd4e-4d5f-883a-ca574f685756.png)

This function should really live in a change handler on the underlying select component, so that it updates the filter state in the parent when the filter state actually changes.

So you could change `checkFilters` in `ColonyMembers` to:

```js

const handleFilterChange = (name, value) => {
  if(filters[name] !== value) {
    setFilters((filters) => ({
         ...filters,
         [name]: value    
      }))
  }
}

```
Then in the `Filter` component call it like:

``` js
<Select
        onChange={(value) => handleFilterChange(name, value)}
        appearance={appearance}
        name={name}
        options={options}
        label={label}
      />

```

Reply:
@willm30 this works however after this change, reset does not correctly reset the filter state. Is there a way to incorporate updating the filter state directly from the reset() function?

Reply:
No, you will need to call `handleFilterChange` with the default values alongside calling reset, since we're holding said values outside the form state. Alternatively (and perhaps this is actually better), you wrap the whole thing in a form instead of just the filters aside, so the Members content can directly access the filter state instead of us duplicating it in component state in `ColonyMembers`